app/youtube/service_test.go

```diff
@@ -83,32 +95,37 @@ func TestService_Do(t *testing.T) {
 
 	res, err = boltStore.Load("channel2", 10)
 	require.NoError(t, err)
-	assert.Equal(t, 2, len(res), "two entries for channel1, skipped duplicate")
-	assert.Equal(t, "vid2", res[0].VideoID)
-	assert.Equal(t, "vid1", res[1].VideoID)
+	assert.Equal(t, 3, len(res), "two entries for channel1, skipped duplicate")
+	assert.Equal(t, "vid3", res[0].VideoID)
+	assert.Equal(t, "vid2", res[1].VideoID)
+	assert.Equal(t, "vid1", res[2].VideoID)
 
-	require.Equal(t, 4, len(downloader.GetCalls()))
+	require.Equal(t, 6, len(downloader.GetCalls()))
 	require.Equal(t, "vid1", downloader.GetCalls()[0].ID)
 	require.True(t, downloader.GetCalls()[0].Fname != "")
 
-	rssData, err := os.ReadFile("/tmp/channel1.xml")
+	rssData, err := os.ReadFile(tempDir + "/channel1.xml") // nolint
 	require.NoError(t, err)
 	t.Logf("%s", string(rssData))
 	assert.Contains(t, string(rssData), "<guid>channel1::vid1</guid>")
 	assert.Contains(t, string(rssData), "<guid>channel1::vid2</guid>")
 	assert.Contains(t, string(rssData), "<itunes:duration>1234</itunes:duration>")
 
-	rssData, err = os.ReadFile("/tmp/channel2.xml")
+	rssData, err = os.ReadFile(tempDir + "/channel2.xml") // nolint
 	require.NoError(t, err)
 	assert.Contains(t, string(rssData), "<guid>channel2::vid1</guid>")
 	assert.Contains(t, string(rssData), "<guid>channel2::vid2</guid>")
 	assert.Contains(t, string(rssData), "<itunes:duration>1234</itunes:duration>")
 
-	require.Equal(t, 4, len(duration.FileCalls()))
-	assert.Equal(t, "/tmp/e4650bb3d770eed60faad7ffbed5f33ffb1b89fa.mp3", duration.FileCalls()[0].Fname)
-	assert.Equal(t, "/tmp/4308c33c7ddb107c2d0c13a905e4c6962001bab4.mp3", duration.FileCalls()[1].Fname)
-	assert.Equal(t, "/tmp/3be877c750abb87daee80c005fe87e7a3f824fed.mp3", duration.FileCalls()[2].Fname)
-	assert.Equal(t, "/tmp/648f79b3a05ececb8a37600aa0aee332f0374e01.mp3", duration.FileCalls()[3].Fname)
+	t.Logf("%v", duration.FileCalls())
+	require.Equal(t, 11, len(duration.FileCalls()))
```

Comment:
I'm not sure why we make 11 calls here; a message/comment explaining why 11 and not the same number as GetCalls will be helpful